docs/devel/e2e-tests.md

```diff
@@ -232,6 +232,102 @@ stale permissions can cause problems.
 
   - `sudo iptables -F`, clear ip tables rules left by the kube-proxy.
 
+### Federation e2e tests
+
+By default, `e2e.go` provisions a single Kubernetes cluster, and any `Feature:Federation` ginkgo tests will be skipped.
+
+Federation e2e testing involve bringing up multiple "underlying" Kubernetes clusters,
+and deploying the federation control plane as a Kubernetes application on the underlying clusters.
+
+The federation e2e tests are still managed via `e2e.go`, but require some extra configuration items.
+
+#### Configuring federation e2e tests
+
+The following environment variables will enable federation e2e building, provisioning and testing.
+
+```sh
+$ export FEDERATION=true
+$ export E2E_ZONES="us-central1-a us-central1-b us-central1-f"
```

Comment:
It might be good to add a note for e2e developers, that they can include just one zone here for faster cluster turn up and down.

Or maybe we can add a separate section at the end saying how to do faster cluster turn up and down.
I use `NUM_NODES=1 E2E_ZONES="us-central1-f"` which brings up one kubernetes cluster with 2 nodes (one master node and one normal node) which is usually enough for whatever I am testing.